### ISSUE TITLE
💡 Allow the choice of autothreading on only messages from a certain user, bot, or role

### ISSUE THREAD

**Issue by DMFriends** [MarcusOtter/discord-needle#61]
**Describe the solution you'd like**
I think it would be nice to be able to choose which user or bot in my server to automatically create threads on.

**Comment by MarcusOtter** [comment]
I think this would be pretty cool, we could add a `mentionable` parameter to `/configure auto-threading` so it could pick users with a certain role or certain users.

**Comment by ririko5834** [comment]
Yes that threads will be created just from messages from some user/role. This should also allow that threads will be made from bot's messages.

**Comment by DMFriends** [comment]
> 

Exactly, that's what I was going for. Roles could actually be cool too, in addition to regular users and bots.

**Comment by DMFriends** [comment]
@MarcusOtter Just wanted to mention that the `include-bots` parameter would probably not be needed if we were to implement this.

Also, in case it isn't clear: I don't know TypeScript very well, but I looked at the `[configure.ts](https://github.com/MarcusOtter/discord-needle/blob/main/src/commands/configure.ts)` and `[configHelpers.ts](https://github.com/MarcusOtter/discord-needle/blob/main/src/helpers/configHelpers.ts)` files, and I have a rough idea of what we would need to do to implement this, but again, I don't know TypeScript well, so I was hoping you would be able to implement this yourself. Thanks again!

**Comment by MarcusOtter** [comment]
I've had a fever for a week but still think this is a good idea :)
It's not of the top of my priorities in this project (and this project is not my top priority among my projects) but I'll happily start looking at it after #60

**Comment by DMFriends** [comment]
> I've had a fever for a week but still think this is a good idea :) It's not of the top of my priorities in this project (and this project is not my top priority among my projects) but I'll happily start looking at it after #60

Oh, that's too bad, hope you are well now :)
That's totally fine, again not urgent. Thanks!

**Comment by RenidJBS** [comment]
This is exactly what I need for my discord server as well. If it could thread in one channel based on mentions and allow bots, this would be perfect!  I have YAGPDB reacting to specific mentions, but I need it to create a thread and it’s unable to. Yours would fill that gap!  Very excited for this feature!

**Comment by DMFriends** [comment]
Hey @MarcusOtter , looks like #60 is done, have you started working on this? (No worries if you haven't.)

**Comment by MarcusOtter** [comment]
No updates on this right now, but it might be closed in favor of #93 in the future as that would cover more use cases (including this one)

**Comment by DMFriends** [comment]
> No updates on this right now, but it might be closed in favor of #93 in the future as that would cover more use cases (including this one)

I see, thanks!

**Comment by MarcusOtter** [comment]
Closing in favor of #93 which will accomplish the same thing
